### PR TITLE
Repeat density subworkflow

### PR DESCRIPTION
**Pull request by DLBPointon**
Updates
- Fixed dev - Uncommented out pipeline code.
- Added Punchlist subworkflow.
    - Output also changed to indicate it is actually punchlist and not genealignment.
- Added Repeat Density subworkflow.
- Added Gap_finder subworkflow.
- Added stubs to all modules and updated versions to match similar nf-core modules.
- Modified config files for all new additions.
- Installed a number of new modules.
- Moved SAMTOOLS faidx outside of nuc_alignemnt subworkflow. Longread also uses the same output, so we have reduced this module being used 5 times to 1.
- Contains workflows to be used for HiC_mapping subworkflow, however not yet integrated.
- Minimap2 config changed to allow for larger gene alignments.
- Updates to the test yaml

NF-core updates:
- Written 3 modules and submitted into nf-core for this pipeline.

## PR checklist

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
  - [X] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/treeval/tree/master/.github/CONTRIBUTING.md)
  - [X] If necessary, also make a PR on the nf-core/treeval _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [X] Make sure your code lints (`nf-core lint`).
- [X] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

All docs will be updated in a sit down meeting.

### CODE REVIEW

**Review by priyanka-surana — COMMENTED** [review]
Since I could not run the test – tried with `s3_test` profile, some generic comments below. Also you might want to set your `outdir` default to something basic like `results`. If a user does not provide a value, it creates a directory named `[:]`. Let me know if I should be testing differently.

File: modules/local/mv_to_sanger.nf
[the remark this answers is not in the document]
Comment (DLBPointon) [reply modules/local/mv_to_sanger.nf:1]: This is now outdated, last week we were going to have this move some punchlist files to a directory. This is now defunct as the punchlists will instead be going along with all the other output. 
I'll remove this in the morning.

File: bin/add_len2gap.sh
[the remark this answers is not in the document]
Comment (DLBPointon) [reply bin/add_len2gap.sh:1]: I thought so, hence why I did this is replace_dots. For some reason, i couldn't get this module to do the same without a script. I assumed it was down to the use of column variables.

i'll have another look in the morning and ping you if not.

File: bin/rename_ids.sh
[the remark this answers is not in the document]
Comment (DLBPointon) [reply bin/rename_ids.sh:1]: I thought so, hence why I did this is replace_dots. For some reason, i couldn't get this module to do the same without a script. I assumed it was down to the use of column variables.

i'll have another look in the morning and ping you if not.

File: modules/local/chunkfasta.nf
[the remark this answers is not in the document]
Comment (DLBPointon) [reply modules/local/chunkfasta.nf:1]: I thought adding a stub was part of their new guidelines? At least for nf-core modules anyway.

If that's the case, then I'd like to make sure all modules are as close to the standard as possible.

[the remark this answers is not in the document]
Comment (priyanka-surana) [reply modules/local/chunkfasta.nf:1]: > I thought adding a stub was part of their new guidelines? At least for nf-core modules anyway.
> 
> If that's the case, then I'd like to make sure all modules are as close to the standard as possible.

It is definitely recommended and it is great you are doing it. There are still some old ones that do not have that feature. https://github.com/nf-core/modules/blob/master/modules/nf-core/busco/main.nf

File: subworkflows/local/gap_finder.nf [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env nextflow
+
+include { SEQTK_CUTN } from '../../modules/nf-core/seqtk/cutn/main'
+include { GAP_LENGTH } from '../../modules/local/gap_length'
+
+workflow GAP_FINDER {
+    take:
+    reference_tuple
```
[the remark this answers is not in the document]
Comment (DLBPointon) [reply subworkflows/local/gap_finder.nf]: This has now been added to all subworkflows.

File: bin/rename_ids.sh
[the remark this answers is not in the document]
Comment (DLBPointon) [reply bin/rename_ids.sh:1]: I've brought it up in a TreeVal meeting, and we would like to have this as a dedicated Phase 3 - reorganising / documentation round.

So would it be possible for the merge please.

[the remark this answers is not in the document]
Comment (priyanka-surana) [reply bin/rename_ids.sh:1]: Does that mean no documentation till phase 3?

**Review by priyanka-surana — APPROVED** [review]
Note: I have not run the tests or done an in-depth review for that reason. 
Approving so the treeval team can merge and move to the next subworkflow.

**Pull request merged**